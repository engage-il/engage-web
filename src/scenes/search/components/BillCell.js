// @flow
import React, { Component } from 'react'
import { createFragmentContainer, graphql } from 'react-relay'
import moment from 'moment'
import { css } from 'glamor'
import type { Rule } from 'glamor'
import type { Bill } from 'shared/types'
import { Button } from 'shared/components'
import { stylesheet, colors, mixins } from 'shared/styles'

let BillCell = class BillCell extends Component {
  props: {
    bill: Bill,
    styles?: Rule
  }

  // lifecycle
  render () {
    const { bill, styles } = this.props
    const date = moment(bill.hearing.date)

    return <div {...css(rules.container, styles)}>
      <div {...rules.info}>
        <div {...rules.header}>
          <div {...rules.document}>
            <h3>{bill.title}</h3>
            <span>{bill.documentNumber}</span>
          </div>
          <p>{date.calendar()}</p>
        </div>
        {bill.summary && <p {...rules.summary}>{bill.summary}</p>}
      </div>
      <div {...rules.actions}>
        <Button
          type='solid'
          styles={rules.button}
          to={bill.witnessSlipUrl}
          label='Take Action'
          iconName='pencil-square-o'
        />
        <Button
          styles={rules.button}
          to={`bill/${bill.id}`}
          label='More Info'
          iconName='file-text-o'
        />
      </div>
    </div>
  }
}

BillCell = createFragmentContainer(BillCell, graphql`
  fragment BillCell_bill on Bill {
    id
    documentNumber
    title
    summary
    witnessSlipUrl
    detailsUrl
    fullTextUrl
    hearing {
      date
    }
  }
`)

const rules = stylesheet({
  container: {
    ...mixins.shadows.low(),
    ...mixins.borders.low(),
    display: 'flex',
    padding: 15,
    marginBottom: 15,
    backgroundColor: colors.neutral,
    '&:last-child': {
      marginBottom: 0
    },
    ...mixins.mobile({
      flexDirection: 'column'
    })
  },
  info: {
    ...mixins.borders.low(['right']),
    flex: 1,
    paddingRight: 15,
    ...mixins.mobile({
      ...mixins.borders.low(['bottom']),
      paddingRight: 0,
      paddingBottom: 15,
      marginBottom: 15,
      borderRight: 'none'
    })
  },
  header: {
    display: 'flex',
    flexDirection: 'column'
  },
  document: {
    marginBottom: 5,
    '> *': {
      display: 'inline-block'
    },
    '> h3': {
      ...mixins.fonts.bold,
      marginRight: 10
    }
  },
  actions: {
    display: 'flex',
    flexDirection: 'column',
    paddingLeft: 15,
    ...mixins.mobile({
      flexDirection: 'row',
      paddingLeft: 0
    })
  },
  button: {
    width: 200,
    marginBottom: 10,
    ':last-child': {
      marginBottom: 0
    },
    ...mixins.mobile({
      flex: 1,
      width: 'auto',
      marginBottom: 0,
      marginRight: 10,
      ':last-child': {
        marginRight: 0
      }
    })
  },
  summary: {
    ...mixins.borders.low(['top']),
    marginTop: 10,
    paddingTop: 10
  }
})

export { BillCell }
