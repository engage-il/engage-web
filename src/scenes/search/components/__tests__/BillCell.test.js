/* eslint-env jest */
import { assign } from 'lodash'
import React from 'react'
import { shallow } from 'enzyme'
import { BillCell } from '../BillCell'

let subject
let props

function loadSubject () {
  subject = shallow(<BillCell {...props} />).dive()
}

function test () {
  expect(subject).toMatchSnapshot()
}

beforeEach(() => {
  props = {
    bill: {
      documentNumber: 'HB1234',
      title: 'Foo',
      summary: 'A bill, fantastic',
      witnessSlipUrl: 'http://www.test.com/slip',
      billDetailsUrl: 'http://www.test.com/details',
      fullTextUrl: 'http://www.test.com/text',
      hearing: {
        date: '2010-01-01T00:00:00-06:00'
      }
    }
  }
  subject = null
})

describe('#render', () => {
  it('renders properly', () => {
    loadSubject()
    test()
  })

  it('renders properly when data is missing', () => {
    props.bill.summary = null
    loadSubject()
    test()
  })
})
