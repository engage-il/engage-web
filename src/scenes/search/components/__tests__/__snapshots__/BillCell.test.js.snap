// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders properly 1`] = `
<div
  data-css-1n4vje7=""
>
  <div
    data-css-154k5mk=""
  >
    <div
      data-css-1t62idy=""
    >
      <div
        data-css-102sxgz=""
      >
        <h3>
          Foo
        </h3>
        <span>
          HB1234
        </span>
      </div>
      <p>
        01/01/2010
      </p>
    </div>
    <p
      data-css-jpl8hl=""
    >
      A bill, fantastic
    </p>
  </div>
  <div
    data-css-1e193l7=""
  >
    <Button
      iconName="pencil-square-o"
      label="Take Action"
      styles={
        Object {
          "data-css-wkh0gw": "",
        }
      }
      to="http://www.test.com/slip"
      type="solid"
    />
    <Button
      iconName="file-text-o"
      label="More Info"
      styles={
        Object {
          "data-css-wkh0gw": "",
        }
      }
      to="bill/undefined"
    />
  </div>
</div>
`;

exports[`#render renders properly when data is missing 1`] = `
<div
  data-css-1n4vje7=""
>
  <div
    data-css-154k5mk=""
  >
    <div
      data-css-1t62idy=""
    >
      <div
        data-css-102sxgz=""
      >
        <h3>
          Foo
        </h3>
        <span>
          HB1234
        </span>
      </div>
      <p>
        01/01/2010
      </p>
    </div>
  </div>
  <div
    data-css-1e193l7=""
  >
    <Button
      iconName="pencil-square-o"
      label="Take Action"
      styles={
        Object {
          "data-css-wkh0gw": "",
        }
      }
      to="http://www.test.com/slip"
      type="solid"
    />
    <Button
      iconName="file-text-o"
      label="More Info"
      styles={
        Object {
          "data-css-wkh0gw": "",
        }
      }
      to="bill/undefined"
    />
  </div>
</div>
`;
